spec/helpers.rb

```diff
@@ -9,4 +9,51 @@ def json
   def auth_headers
     user.create_new_auth_token
   end
+
+  def retrieve_notifications_mock
```

Comment:
I would move these methods to another place. For example to a class in `spec/support/notifications_mocker.rb`